### PR TITLE
style: move the see full results to the bottom

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Style

This PR solves [FX-4650]

|Before|After|
|---|---|
|image|image|

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4650]: https://artsyproduct.atlassian.net/browse/FX-4650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Search/SearchBar.tsx [hunk 1]
```diff
@@ -18,10 +18,13 @@ import { Router } from "found"
 import { isEmpty } from "lodash"
 import { throttle } from "lodash"
 import qs from "qs"
+// eslint-disable-next-line no-restricted-imports
 import Autosuggest from "react-autosuggest"
 import { RelayRefetchProp, createRefetchContainer, graphql } from "react-relay"
+// eslint-disable-next-line no-restricted-imports
 import { data as sd } from "sharify"
 import styled from "styled-components"
+// eslint-disable-next-line no-restricted-imports
```
Comment (rajsam003) [inline src/Components/Search/SearchBar.tsx:27]: I know it's not an ideal solution. Any better solution?

File: src/Components/Search/Suggestions/SuggestionItem.tsx [hunk 2]
```diff
@@ -98,7 +105,13 @@ const DefaultSuggestion: React.FC<SuggestionItemProps> = ({
   const matches = match(display, query)
   const parts = parse(display, matches)
   const partTags = parts.map(({ highlight, text }, index) =>
-    highlight ? <strong key={index}>{text}</strong> : text
+    highlight ? (
+      <strong key={index} style={{ color: "blue" }}>
```
Comment (rajsam003) [inline src/Components/Search/Suggestions/SuggestionItem.tsx:109]: `<strong color="blue100">` tag is not working.  
I can replace it with <Text> but it's not simple like strong. Also, the alignments change. 
image

File: src/Components/Search/SearchBar.tsx [hunk 3]
```diff
@@ -18,10 +18,11 @@ import { Router } from "found"
 import { isEmpty } from "lodash"
 import { throttle } from "lodash"
 import qs from "qs"
+// eslint-disable-next-line no-restricted-imports
 import Autosuggest from "react-autosuggest"
 import { RelayRefetchProp, createRefetchContainer, graphql } from "react-relay"
-import { data as sd } from "sharify"
 import styled from "styled-components"
+// eslint-disable-next-line no-restricted-imports
```
Comment (dimatretyak) [inline src/Components/Search/SearchBar.tsx:25]: **praise**: thank for these changes 🙏

File: src/Components/Search/SearchBar.tsx [hunk 4]
```diff
@@ -410,7 +412,7 @@ export class SearchBar extends Component<Props, State> {
     // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
     const edges = get(viewer, v => v.searchConnection.edges, [])
     // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-    const suggestions = [firstSuggestionPlaceholder, ...edges]
+    const suggestions = [...edges, firstSuggestionPlaceholder]
```
Comment (dimatretyak) [inline src/Components/Search/SearchBar.tsx:415]: **question**: What do you think about renaming `firstSuggestionPlaceholder`? For example, in `lastSuggestionPlaceholder` or even better in `seeAllResultsPlaceholder`

Comment (dimatretyak) [reply src/Components/Search/SearchBar.tsx:415]: Also, I would rename all the related code (for example, as [here](https://github.com/artsy/force/blob/056073747177a3b77a2dd5bbbb068760dbb9b0ae/src/Components/Search/SearchBar.tsx#L341), [here](https://github.com/artsy/force/blob/056073747177a3b77a2dd5bbbb068760dbb9b0ae/src/Components/Search/SearchBar.tsx#L408) and [here](https://github.com/artsy/force/blob/056073747177a3b77a2dd5bbbb068760dbb9b0ae/src/Components/Search/Suggestions/SuggestionItem.tsx#L32))

File: src/Components/Search/Suggestions/SuggestionItem.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/Components/Search/Suggestions/SuggestionItem.tsx:109]: we can create the new styled component. something like this

```javascript
const Highlight = styled.strong`
  color: ${themeGet("colors.blue100")};
`
...
const partTags = parts.map(({ highlight, text }, index) =>
  highlight ? <Highlight key={index}>{text}</Highlight> : text
)
```

File: src/Components/Search/Suggestions/SuggestionItem.tsx [hunk 5]
```diff
@@ -40,12 +41,18 @@ export const FirstSuggestionItem: React.FC<SuggestionItemProps> = ({
...
+      mt={1}
+      borderTop="1px solid"
+      borderTopColor="black10"
       onClick={handleClick}
       to={href}
     >
-      <Text variant="sm">See full results for &ldquo;{query}&rdquo;</Text>
+      <Flex alignItems="center">
+        <Text variant="sm" mr={1}>
+          See full results for &ldquo;{query}&rdquo;
```
Comment (dimatretyak) [inline src/Components/Search/Suggestions/SuggestionItem.tsx:52]: we don't need to use `&ldquo;` and `&rdquo;` here (Just `See full results for {query}`)

Comment (dimatretyak) [reply src/Components/Search/Suggestions/SuggestionItem.tsx:52]: Also i think we need to highlight query (see screenshot)

for-sam

File: src/Components/Search/Suggestions/SuggestionItem.tsx [hunk 6]
```diff
@@ -40,12 +41,18 @@ export const FirstSuggestionItem: React.FC<SuggestionItemProps> = ({
...
+        </Text>
+        <ChevronButton />
```
Comment (dimatretyak) [inline src/Components/Search/Suggestions/SuggestionItem.tsx:54]: **suggestion**: I think it's better to use `ChevronRightIcon` here. Something like this

```javascript
import ChevronRightIcon from "@artsy/icons/ChevronRightIcon"
...
<ChevronRightIcon height="14px" />
```

File: src/Components/Search/SearchBar.tsx (hunk 4 above)
Comment (rajsam003) [reply src/Components/Search/SearchBar.tsx:415]: Good idea

File: src/Components/Search/SearchBar.tsx [hunk 7]
```diff
@@ -400,19 +400,19 @@ export class SearchBar extends Component<Props, State> {
       value: term,
     }
 
-    const firstSuggestionPlaceholder = {
+    const seeAllResultsPlaceholder = {
       node: {
         displayLabel: term,
-        displayType: "FirstItem",
+        displayType: "SeeFullResultsSuggestionItem",
```
Comment (dblandin) [inline src/Components/Search/SearchBar.tsx:406]: **suggestion/blocking**: I think this has the risk of altering some analytics to track when a user has clicked on the "See full results" suggestion. For now, can we keep the displayType as `FirstItem` until we're able to touch base with @abhitip and align on a new value?

**Review by dblandin — CHANGES_REQUESTED** [review]
LGTM overall but I left one blocking bit of feedback.

File: src/Components/Search/SearchBar.tsx (hunk 7 above)
Comment (rajsam003) [reply src/Components/Search/SearchBar.tsx:406]: It makes sense. I am reverting back

Comment (dimatretyak) [reply src/Components/Search/SearchBar.tsx]: > reverting back

I think we can rename it in other places, but leave only the old value (`FirstItem`) here

**Review by dimatretyak — APPROVED** [review]
LGTM. Thanks for changes 🙏

File: src/Components/Search/SearchBar.tsx (hunk 7 above)
Comment (dimatretyak) [reply src/Components/Search/SearchBar.tsx]: > I think this has the risk of altering some analytics to track when a user has clicked on the "See full results" suggestion

Yes, you're right

Sources - force-staging - Segment 2023-03-15 13-53-23

Comment (dblandin) [reply src/Components/Search/SearchBar.tsx]: Yeah, exactly. The code variables can be renamed/refactored as long as the analytics event remains consistent.

**Pull request merged**